configserver/src/main/java/com/yahoo/vespa/config/server/application/ZKTenantApplications.java

```diff
@@ -161,6 +164,11 @@ public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) th
             case CHILD_REMOVED:
                 applicationRemoved(ApplicationId.fromSerializedForm(Path.fromString(event.getData().getPath()).getName()));
                 break;
+            default:
```

Comment:
This includes connection loss, connection reconnect etc. There should be a full resync on connection reconnect since lots of events may have been lost. Why is removeApplications() not handling additions of applications?

CHILD_ADDED
CHILD_REMOVED
CHILD_UPDATED
CONNECTION_LOST
CONNECTION_RECONNECTED
CONNECTION_SUSPENDED
INITIALIZED

Reply:
removeApplications() only removes applications because there is nothing that adds an application in this code (CHILD_ADDED calls applicationAdded(), but it only logs, which confused me too). An application is added when a session is activated (see makeActive() in RemoteSession).

I agree that we need a full resync on reconnect, that's what I want to do. This PR was only to get more logging, but I wanted info from ApplicationMapper, so I made removeApplications()

Reply:
OK